src/execute.rs

```diff
@@ -422,7 +432,38 @@ pub fn run(cli: Cli) -> Result<(), Box<dyn Error>> {
             ProofSystem::KZG => {
                 let proof = Snark::load::<KZGCommitmentScheme<Bn256>>(&proof_path, None, None)?;
                 let code = DeploymentCode::load(&deployment_code_path)?;
-                evm_verify(code, proof)?;
+                evm_verify(code, proof.clone())?;
+
+                if sol_code_path.is_some() {
+                    let (anvil, client) = setup_eth_backend().await;
+
+                    let compiled = Solc::default().compile_source(sol_code_path.unwrap())?;
+                    let (abi, bytecode, _runtime_bytecode) =
+                        compiled.find("Verifier").expect("could not find contract").into_parts_or_default();
+                    let factory = ContractFactory::new(abi, bytecode, client.clone());
+                    let contract = factory.deploy(())?.send().await?;
+                    let addr = contract.address();
+
+                    abigen!(Verifier, "./Verifier.json");
+                    let contract = Verifier::new(addr, client.clone());
+
+                    let mut public_inputs = vec![];
+                    for val in &proof.instances[0] {
+                        let bytes = val.to_repr();
+                        let u = U256::from_little_endian(bytes.as_slice());
+                        public_inputs.push(u);
+                    }
+
+                    let result = contract.verify(
+                        public_inputs,
+                        ethers::types::Bytes::from(proof.proof.to_vec()),
+                    ).call().await.unwrap();
+
+                    info!("Solidity verification result: {}", result);
+                    assert!(result);
```

Comment:
maybe box this up in a similar fashion so we don't get hard failures if used by `pyezkl`. This could for instance be similar to the error handling in the `evm_verify` function: 

```rust
 if result.reverted {
            return Err(Box::new(EvmVerificationError::Reverted));
        }
```

Though I'm not sure this result object will be the same -- so the above might have to be tailored to the returned object

Reply:
Ditto, check out https://github.com/zkonduit/ezkl/commit/d6abd3f6b5614364437a8cd83f25a1c903f8edfd#diff-b9cde71f35f4266fc72cee1dccc59f632255fa628e923d8cbd9c6429b7a1dcf6R75

`return Err(Box::new(EvmVerificationError::InvalidProof));`